### PR TITLE
Account Management security

### PR DESCRIPTION
**Pull request by weskubo-cgi**
Added security to Manage Users and Manage Organization Facilities.
Made the following updates which aren't explicit in the Portal Roles matrix:
- Removed Account Maintenance option from New Message if not Account Manager
- Did not secure View Org/Facility Information as all roles have this permission
- Removed the ability to assign Impersonate as a role
- Updated sorting rules in Manage Users View from 'Account Management' to 'Account Manager' but need to revisit this requirement
- There is a slight discrepancy between the Consolidated and Expanded matrix in that we have lost the ability for Impersonate to View Users. This will be handled as a separate ticket.

### CODE REVIEW

**Review by jgstorey — APPROVED** [review]
Confirming reviewed.

File: frontend/src/views/account-mgmt/ManageUsersView.vue [hunk 1]
```diff
@@ -211,8 +207,9 @@ export default {
       if (!users) return []
       return users.sort((a, b) => {
         // Check for account management role and sort by it, with true values first
-        const roleA = a.role === this.ROLES.ACCOUNT_MANAGEMENT
-        const roleB = b.role === this.ROLES.ACCOUNT_MANAGEMENT
+        // TODO (weskubo-cgi) Revisit this requirement given new Roles Security Matrix
+        const roleA = a.role === this.ROLES.ACCOUNT_MANAGER
```
Comment (vietle-cgi) [inline frontend/src/views/account-mgmt/ManageUsersView.vue:211]: since role is an object, we will need to update this line to a.role?.roleName

Comment (vietle-cgi) [reply frontend/src/views/account-mgmt/ManageUsersView.vue:211]: same for b.role

File: frontend/src/components/account-mgmt/ManageUserDialog.vue [hunk 2]
```diff
@@ -220,11 +224,13 @@ export default {
         this.userOperationType = Object.keys(value).length === 0 ? 'add' : 'update'
         this.selectedFacilityIds = value.facilities
         this.user = value
+        if (!this.user.role) this.user.role = {}
       },
     },
   },
   async created() {
     this.facilitiesToAdminister = await this.getUserFacilities(this.userInfo.contactId, true)
+    this.ROLES = ROLES
```
Comment (vietle-cgi) [inline frontend/src/components/account-mgmt/ManageUserDialog.vue:233]: I don't think we need this one.

**Review by vietle-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/views/applications/ApplicationView.vue [hunk 3]
```diff
@@ -80,7 +80,6 @@ export default {
   },
 
   computed: {
-    ...mapState(useAppStore, ['getRoleNameById']),
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/applications/ApplicationView.vue:83]: Removed.

File: frontend/src/components/messages/NewRequestDialog.vue [hunk 4]
```diff
@@ -464,20 +471,20 @@ export default {
...
         this.resetModelData(isAccountMaintenance)
       },
     },
-
   },
   created() {
     this.PHONE_FORMAT = PHONE_FORMAT
     this.EMAIL_FORMAT = EMAIL_FORMAT
     this.AUTO_REPLY_MESSAGE = 'Your change request is complete.'
+    this.CATEGORY_ACCCOUNT_MAINTENANCE = 'Account Maintenance'
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/components/messages/NewRequestDialog.vue:487]: Yeah I didn't even see that.

File: frontend/src/views/account-mgmt/ManageOrganizationView.vue [hunk 5]
```diff
@@ -117,20 +114,15 @@ export default {
   computed: {
     ...mapState(useAuthStore, ['userInfo']),
     ...mapState(useAppStore, ['getRequestCategoryIdByName']),
-
-    isAccountManager() {
-      return this.hasRole(this.ROLES.ACCOUNT_MANAGEMENT)
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/account-mgmt/ManageOrganizationView.vue:122]: Removed.

File: frontend/src/views/account-mgmt/ManageUsersView.vue [hunk 6]
```diff
@@ -150,6 +147,7 @@ export default {
     },
   },
   async created() {
+    this.ROLES = ROLES
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/account-mgmt/ManageUsersView.vue:150]: Removed.

File: frontend/src/views/account-mgmt/ManageUsersView.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/account-mgmt/ManageUsersView.vue:211]: Updated.

File: frontend/src/stores/app.js [hunk 7]
```diff
@@ -30,7 +28,7 @@ export const useAppStore = defineStore('app', {
   getters: {
     getRoleNameById: (state) => {
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/stores/app.js:29]: I'm going to leave it for now but fix the property.

File: frontend/src/views/account-mgmt/AccountMgmtView.vue [hunk 8]
```diff
@@ -22,14 +22,15 @@
...
+import AppBackButton from '@/components/ui/AppBackButton.vue'
+import permissionsMixin from '@/mixins/permissionsMixin'
 import { useAuthStore } from '@/stores/auth'
 
 export default {
   components: { AppBackButton },
-  mixins: [rolesMixin],
+  mixins: [permissionsMixin],
   methods: {
     ...mapActions(useAuthStore, ['hasRole']),
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/account-mgmt/AccountMgmtView.vue:35]: Updated and remove store getter.

File: frontend/src/components/account-mgmt/ManageUserDialog.vue (hunk 2 above)
Comment (weskubo-cgi) [reply frontend/src/components/account-mgmt/ManageUserDialog.vue:233]: Removed.

File: frontend/src/views/account-mgmt/ManageUsersView.vue (hunk 1 above)
Comment (weskubo-cgi) [reply frontend/src/views/account-mgmt/ManageUsersView.vue:211]: Updated and re-tested.

**Review by vietle-cgi — APPROVED** [review]

**Pull request merged**